mono/mini/interp/interp.c

```diff
@@ -153,6 +162,7 @@ frame_stack_alloc_ovf (FrameStack *stack, int size, StackFragment **out_frag)
 		current = stack->current = current->next;
 		current->pos = (guint8*)&current->data;
 	} else {
+		free_frag (current->next);
```

Comment:
Is there a preemptive GC problem here? `interp_mark_stack` scans while `frag` is non-null, following `frag->next` links.  If you free `current->next` without setting it to NULL, I think there's a period of time where the GC could be scanning deallocated memory.

Reply:
Good point, thank you. For what it's worth, the case is quite uncommon to actually happen (= `current->next` is set _and_ the requested frame size wouldn't fit).  In combination with GC, I'm sure that could lead to funny bugs... 🙂